internals-js/src/specs/sourceSpec.ts

```diff
@@ -140,8 +144,271 @@ export class SourceSpecDefinition extends FeatureDefinition {
   sourceFieldDirective(schema: Schema) {
     return this.directive<SourceFieldDirectiveArgs>(schema, 'sourceField')!;
   }
+
+  override validateSubgraphSchema(schema: Schema): GraphQLError[] {
+    const sourceAPIInSchema = schema.directive('sourceAPI') as DirectiveDefinition<SourceAPIDirectiveArgs> | undefined;
+    const sourceTypeInSchema = schema.directive('sourceType') as DirectiveDefinition<SourceTypeDirectiveArgs> | undefined;
+    const sourceFieldInSchema = schema.directive('sourceField') as DirectiveDefinition<SourceFieldDirectiveArgs> | undefined;
+
+    if (!(sourceAPIInSchema || sourceTypeInSchema || sourceFieldInSchema)) {
+      // If none of the @source* directives are present, nothing needs
+      // validating.
+      return [];
+    }
+
+    const errors: GraphQLError[] = [];
+    const apiNameToProtocol = new Map<string, ProtocolName>();
+
+    if (sourceAPIInSchema) {
+      sourceAPIInSchema.applications().forEach(application => {
+        const { name, ...rest } = application.arguments();
```

Comment:
Is there a reason you're doing this as opposed to `const { name, http }`?

Reply:
Just trying to be agnostic to eventually having more than just `http` in `KNOWN_SOURCE_PROTOCOLS`.

Reply:
It's just style, so I'm not going to hold up the release for it, but imo it's a little cleaner to say

```typescript
const { api, http, grpc, sql } = application.arguments();
if (http) {
  assert(apiNameToProtocol.get(api) === 'http', ...);
} else if (grpc) {
  assert(apiNameToProtocol.get(api) === 'grpc', ...);
...
}
```